google-cloud-firestore/lib/google/cloud/firestore/collection.rb

```diff
@@ -0,0 +1,663 @@
+          #
+          #   # Get a document reference with data
+          #   random_ref = cities_col.add({ name: "New York City" })
+          #
+          #   # The document ID is randomly generated
+          #   random_ref.document_id #=> "RANDOMID123XYZ"
+          #
+          def add data = nil
+            data ||= {}
+            doc.tap { |d| d.create data }
+          end
+
+          ##
+          # Retrieves a list of documents with data.
+          #
+          # @param [String, Document::Reference] docs One or more strings
+          #   representing the path of the document, or document reference
+          #   objects.
+          # @param [Array<String|Symbol>, String|Symbol] mask A list of field
+          #   paths to filter the returned document data by.
+          #
+          # @yield [documents] The block for accessing the documents with data.
+          # @yieldparam [Document::Snapshot] document A document with data.
+          #
+          # @return [Enumerator<Document::Snapshot>] documents with data list.
+          #
+          # @example
+          #   require "google/cloud/firestore"
+          #
+          #   firestore = Google::Cloud::Firestore.new
+          #
+          #   # Get a collection reference
+          #   cities_col = firestore.col "cities"
+          #
+          #   # Get and print city documents
+          #   cities = ["cities/NYC", "cities/SF", "cities/LA"]
+          #   cities_col.get_all(cities).each do |city|
+          #     puts "#{city.document_id} has #{city[:population]} residents."
+          #   end
+          #
+          def get_all *docs, mask: nil, &block
+            ensure_context!
+
+            # Get document_path, not path
+            doc_paths = Array(docs).flatten.map do |doc_path|
+              if doc_path.respond_to? :document_path
+                doc_path.document_path
+              else
+                doc(doc_path).document_path
+              end
+            end
+
+            context.get_all(doc_paths, mask: mask, &block)
+          end
+          alias_method :get_docs, :get_all
+          alias_method :get_documents, :get_all
+          alias_method :find, :get_all
+
+          ##
+          # Creates a query object.
+          #
+          # @return [Query] A query.
+          #
+          # @example
+          #   require "google/cloud/firestore"
+          #
+          #   firestore = Google::Cloud::Firestore.new
+          #
+          #   # Get a collection reference
+          #   cities_col = firestore.col "cities"
+          #
+          #   # Create a query
+          #   query = cities_col.query.select(:population).from(:cities)
+          #
+          #   # Get/run a query
+          #   query.get do |city|
+          #     puts "#{city.document_id} has #{city[:population]} residents."
+          #   end
+          #
+          def query
+            ensure_context!
+
+            Query.start(parent_path, context).from(collection_id)
```

Comment:
Note that in other APIs this relationship is reversed. A CollectionReference is a subclass of a Query.